docs/feature/search/index.md

```diff
@@ -0,0 +1,255 @@
+(fts)=
+(fulltext)=
+(full-text)=
+
+# Full-Text Search
+
+:::{include} /_include/links.md
+:::
+:::{include} /_include/styles.html
+:::
+
+**BM25 term search based on Apache Lucene, using SQL: CrateDB is all you need.**
+
+:::::{grid}
+:padding: 0
+
+::::{grid-item}
+:class: rubric-slim
+:columns: auto 9 9 9
+
+
+:::{rubric} Overview
+:::
+CrateDB can be used as a database to conduct full-text search operations
+building upon the power of Apache Lucene.
+
+CrateDB is an exceptional choice for handling complex queries and large-scale
+data sets. One of its standout features are its full-text search capabilities,
+built on top of the powerful Lucene library. This makes it a great fit for
+organizing, searching, and analyzing extensive datasets.
+
+:::{rubric} About
+:::
+[Full-text search] leverages the [BM25] search ranking algorithm, effectively
+implementing the storage and retrieval parts of a [search engine].
+
+In information retrieval, Okapi BM25 (BM is an abbreviation of best matching)
+is a ranking function used by search engines to estimate the relevance of
+documents to a given search query.
+::::
+
+
+::::{grid-item}
+:class: rubric-slim
+:columns: auto 3 3 3
+
+```{rubric} Reference Manual
+```
+- [](inv:crate-reference#sql_dql_fulltext_search)
+- [](inv:crate-reference#fulltext-indices)
+- [](inv:crate-reference#ref-create-analyzer)
+
+```{rubric} Related
+```
+- {ref}`sql`
+- {ref}`vector`
+- {ref}`machine-learning`
+- {ref}`query`
+
+{tags-primary}`SQL`
+{tags-primary}`Full-Text Search`
+{tags-primary}`Okapi BM25`
+::::
+
+:::::
+
+
+:::{rubric} Details
+:::
+CrateDB uses Lucene as a storage layer, so it inherits the implementation
+and concepts of Lucene, in the same spirit as Elasticsearch.
+The now popular BM25 method has become the default scoring formula in Lucene
+and is the scoring formula used by CrateDB.
+
+BM25 stands for "Best Match 25", the 25th iteration of this scoring algorithm.
+The excellent article [BM25: The Next Generation of Lucene Relevance] compares
+classic TF/IDF to [Okapi BM25], including illustrative graphs.
+To learn more details about what's inside, please also refer to [Similarity in
+Elasticsearch] and [BM25 vs. Lucene Default Similarity].
+
+:::{div}
+While Elasticsearch uses a [query DSL based on JSON], in CrateDB, you can work
+with text search using SQL.
+:::
+
+
+## Synopsis
+
+Store and query word embeddings using similarity search based on Cosine
+distance.
```

Comment:
CrateDB uses `VectorSimilarityFunction.EUCLIDEAN`, not COSINE.

-- https://github.com/crate/crate/blob/5.6.3/server/src/main/java/io/crate/types/FloatVectorType.java#L55